modules/shadertools/src/modules/picking/picking.js

```diff
@@ -48,44 +41,65 @@ const float COLOR_SCALE = 1. / 255.;
 bool isVertexPicked(vec3 vertexColor) {
   return
     picking_uSelectedColorValid &&
-    abs(vertexColor.r - picking_uSelectedColor.r) < picking_uThreshold &&
-    abs(vertexColor.g - picking_uSelectedColor.g) < picking_uThreshold &&
-    abs(vertexColor.b - picking_uSelectedColor.b) < picking_uThreshold;
+    abs(vertexColor.r - picking_uSelectedColor.r) < 0.001 &&
+    abs(vertexColor.g - picking_uSelectedColor.g) < 0.001 &&
+    abs(vertexColor.b - picking_uSelectedColor.b) < 0.001;
 }
 
 void picking_setPickingColor(vec3 pickingColor) {
-  // Do the comparison with selected item color in vertex shader as it should mean fewer compares
-  picking_vRGBcolor_Aselected.a =
-    float(isVertexPicked(pickingColor));
+  if (picking_uAttribute) {
+    // Use alpha as the validity flag. If pickingColor is [0, 0, 0] fragment is non-pickable
+    picking_vRGBcolor_Aselected.a = step(0.001, length(pickingColor));
```

Comment:
You can do `dot(pickingColor, vec3(1.0))` instead of `length(pickingColor)` here. Avoid the sqrt.